ballerina/rag.bal

```diff
@@ -14,6 +14,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
+isolated Wso2ModelProvider? defaultModelProvider = ();
```

Comment:
Can't we make this final and initialize in the `init` function instead?

Reply:
If that's so, users should always have the configs. Otherwise, it will throw an error.

Specially, when they don't need this API.
wdyt?

Reply:
Why would there be a panic? The type would still be `Wso2ModelProvider?`, just that if the config is provided, it is initialized at init time rather than on the first call to `getDefaultModelProvider `.

Reply:
This is what we did in the np module also.

Reply:
I agree with that approach. 

However, shouldn't we initialize this only when the user actually calls the `getDefaultModelProvider` API? 

Imo, we shouldn't store the default client in memory unnecessarily if it's not being used.